cscs-checks/libraries/gridtools/gridtools_base.py

```diff
@@ -0,0 +1,180 @@
+                                         '-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON']
+
+
+        self.sanity_patterns = sn.assert_found(r'PASSED', self.stdout)
+        self.perf_patterns = {
+            'perf': sn.extractsingle(r'(?P<timer>\w+) ms total',
+                                     self.stdout, 'timer', int)
+        }
+        self.build_system.max_concurrency = 2
+        self.tags = {'production'}
+        self.maintainers = ['CB']
+
+@rfm.parameterized_test(['VerticalAdvectionNaive'], ['VerticalAdvectionMC'], 
+                        ['SimpleHoriDiffNaive'], ['SimpleHoriDiffMC'])
+class GridToolsCheckCPU(GridToolsCheckBase):
+    def __init__(self, test_name):
+        super().__init__()
+        self.descr = 'GridTools host test'
+
+        self.valid_systems = ['daint:mc', 'daint:gpu']
+        self.num_gpus_per_node = 0
+        self.num_tasks = 1
+
+        if test_name == 'VerticalAdvectionNaive':
+            self.build_system.make_opts = ['vertical_advection_dycore_naive']
+            self.executable = 'regression/vertical_advection_dycore_naive'
+            self.executable_opts = ['150', '150', '150']
+            self.reference = {
+                'daint:mc': {
+                    'perf': (3400, None, 0.1)
+                },
+                'daint:gpu': {
+                    'perf': (3800, None, 0.1)
+                }
+            }
+        elif test_name == 'VerticalAdvectionMC':
+            self.build_system.make_opts = ['vertical_advection_dycore_mc']
+            self.executable = 'regression/vertical_advection_dycore_mc'
+            self.executable_opts = ['150', '150', '150']
+            self.reference = {
+                'daint:mc': {
+                    'perf': (3500, None, 0.1)
+                },
+                'daint:gpu': {
+                    'perf': (3700, None, 0.1)
+                }
+        }
+        elif test_name == 'SimpleHoriDiffNaive':
+            self.build_system.make_opts = ['simple_hori_diff_naive']
+            self.executable = 'regression/simple_hori_diff_naive'
+            self.executable_opts = ['100', '100', '100']
+            self.reference = {
+                'daint:mc': {
+                    'perf': (3200, None, 0.1)
+                },
+                'daint:gpu': {
+                    'perf': (3700, None, 0.1)
+                }
+        }
+        elif test_name == 'SimpleHoriDiffMC':
+            self.build_system.make_opts = ['simple_hori_diff_mc']
+            self.executable = 'regression/simple_hori_diff_mc'
+            self.executable_opts = ['100', '100', '100']
+            self.reference = {
+                'daint:mc': {
+                    'perf': (3300, None, 0.1)
+                },
+                'daint:gpu': {
+                    'perf': (3700, None, 0.1)
+                }
+        }
+
+@rfm.parameterized_test(['VerticalAdvectionCuda'], ['SimpleHoriDiffCuda'])
+class GridToolsCheckGPU(GridToolsCheckBase):
```

Comment:
I think the CPU/GPU can be also a parameter. The differences are really small and you can write your test in a nice way in any case with not so many `if`s, especially if you follow my suggestion about not using camel notation in the test type parameters.